In [27]:
import pandas as pd

df = pd.read_csv(r'C:/Users/giihv/Documents/Project Group/bank-additional-full.csv', sep=';')
df

age          job  marital            education  default housing loan  \
0       56    housemaid  married             basic.4y       no      no   no   
1       57     services  married          high.school  unknown      no   no   
2       37     services  married          high.school       no     yes   no   
3       40       admin.  married             basic.6y       no      no   no   
4       56     services  married          high.school       no      no  yes   
...    ...          ...      ...                  ...      ...     ...  ...   
41183   73      retired  married  professional.course       no     yes   no   
41184   46  blue-collar  married  professional.course       no      no   no   
41185   56      retired  married    university.degree       no     yes   no   
41186   44   technician  married  professional.course       no      no   no   
41187   74      retired  married  professional.course       no     yes   no   

         contact month day_of_week  ...  campaign  pdays  previous  \
0      telephone   may         mon  ...         1    999         0   
1      telephone   may         mon  ...         1    999         0   
2      telephone   may         mon  ...         1    999         0   
3      telephone   may         mon  ...         1    999         0   
4      telephone   may         mon  ...         1    999         0   
...          ...   ...         ...  ...       ...    ...       ...   
41183   cellular   nov         fri  ...         1    999         0   
41184   cellular   nov         fri  ...         1    999         0   
41185   cellular   nov         fri  ...         2    999         0   
41186   cellular   nov         fri  ...         1    999         0   
41187   cellular   nov         fri  ...         3    999         1   

          poutcome emp.var.rate  cons.price.idx  cons.conf.idx  euribor3m  \
0      nonexistent          1.1          93.994          -36.4      4.857   
1      nonexistent          1.1          93.994          -36.4      4.857   
2      nonexistent          1.1          93.994          -36.4      4.857   
3      nonexistent          1.1          93.994          -36.4      4.857   
4      nonexistent          1.1          93.994          -36.4      4.857   
...            ...          ...             ...            ...        ...   
41183  nonexistent         -1.1          94.767          -50.8      1.028   
41184  nonexistent         -1.1          94.767          -50.8      1.028   
41185  nonexistent         -1.1          94.767          -50.8      1.028   
41186  nonexistent         -1.1          94.767          -50.8      1.028   
41187      failure         -1.1          94.767          -50.8      1.028   

       nr.employed    y  
0           5191.0   no  
1           5191.0   no  
2           5191.0   no  
3           5191.0   no  
4           5191.0   no  
...            ...  ...  
41183       4963.6  yes  
41184       4963.6   no  
41185       4963.6   no  
41186       4963.6  yes  
41187       4963.6   no  

[41188 rows x 21 columns]

In [59]:
df_sample = df.sample(5)
df_sample

age            job  marital            education default  housing  \
24230   51   entrepreneur  married    university.degree      no      yes   
33379   31     unemployed  married  professional.course      no  unknown   
19864   29     technician  married  professional.course      no      yes   
29719   31  self-employed  married    university.degree      no      yes   
4486    41     technician  married  professional.course      no       no   

          loan    contact month day_of_week  duration  campaign  pdays  \
24230       no   cellular   nov         mon        89         1    999   
33379  unknown   cellular   may         tue       356         1    999   
19864       no   cellular   aug         fri       112         1    999   
29719       no   cellular   apr         mon       291         1    999   
4486        no  telephone   may         tue       148         2    999   

       previous     poutcome  emp.var.rate  cons.price.idx  cons.conf.idx  \
24230         0  nonexistent          -0.1          93.200          -42.0   
33379         0  nonexistent          -1.8          92.893          -46.2   
19864         0  nonexistent           1.4          93.444          -36.1   
29719         0  nonexistent          -1.8          93.075          -47.1   
4486          0  nonexistent           1.1          93.994          -36.4   

       euribor3m  nr.employed   y  
24230      4.191       5195.8  no  
33379      1.291       5099.1  no  
19864      4.966       5228.1  no  
29719      1.405       5099.1  no  
4486       4.856       5191.0  no

In [29]:
df.isnull().sum()

age               0
job               0
marital           0
education         0
default           0
housing           0
loan              0
contact           0
month             0
day_of_week       0
duration          0
campaign          0
pdays             0
previous          0
poutcome          0
emp.var.rate      0
cons.price.idx    0
cons.conf.idx     0
euribor3m         0
nr.employed       0
y                 0
dtype: int64

There's no NaN values

In [30]:
df['pdays'].value_counts()

999    39673
3        439
6        412
4        118
9         64
2         61
7         60
12        58
10        52
5         46
13        36
11        28
1         26
15        24
14        20
8         18
0         15
16        11
17         8
18         7
22         3
19         3
21         2
25         1
26         1
27         1
20         1
Name: pdays, dtype: int64

pdays = number of days that passed by after the client was last contacted from a previous campaign (numeric; 999 means client was not previously contacted)

Almost all of the 'pdays' values are 999

(999  =  39673)

In [36]:
df['previous'].value_counts()

0    35563
1     4561
2      754
3      216
4       70
5       18
6        5
7        1
Name: previous, dtype: int64

In [37]:
df['poutcome'].value_counts()

nonexistent    35563
failure         4252
success         1373
Name: poutcome, dtype: int64

We have 41188 rows, more than 85% of the values of there columns are the same. We need to check if it's worth it using these columns or not!

In [41]:
df['y'].value_counts()

no     36548
yes     4640
Name: y, dtype: int64

Even though more than 85% of this column has the value "no", we cannot drop it. Because it's the most important for our model prediction

Has the client subscribed a term deposit? (binary: 'yes','no')

In [68]:
pd.set_option('display.max_columns', None)
df.head()

age        job  marital    education  default housing loan    contact  \
0   56  housemaid  married     basic.4y       no      no   no  telephone   
1   57   services  married  high.school  unknown      no   no  telephone   
2   37   services  married  high.school       no     yes   no  telephone   
3   40     admin.  married     basic.6y       no      no   no  telephone   
4   56   services  married  high.school       no      no  yes  telephone   

  month day_of_week  duration  campaign  pdays  previous     poutcome  \
0   may         mon       261         1    999         0  nonexistent   
1   may         mon       149         1    999         0  nonexistent   
2   may         mon       226         1    999         0  nonexistent   
3   may         mon       151         1    999         0  nonexistent   
4   may         mon       307         1    999         0  nonexistent   

   emp.var.rate  cons.price.idx  cons.conf.idx  euribor3m  nr.employed   y  
0           1.1          93.994          -36.4      4.857       5191.0  no  
1           1.1          93.994          -36.4      4.857       5191.0  no  
2           1.1          93.994          -36.4      4.857       5191.0  no  
3           1.1          93.994          -36.4      4.857       5191.0  no  
4           1.1          93.994          -36.4      4.857       5191.0  no

In [69]:
df.dtypes

age                 int64
job                object
marital            object
education          object
default            object
housing            object
loan               object
contact            object
month              object
day_of_week        object
duration            int64
campaign            int64
pdays               int64
previous            int64
poutcome           object
emp.var.rate      float64
cons.price.idx    float64
cons.conf.idx     float64
euribor3m         float64
nr.employed       float64
y                  object
dtype: object

The value types seem to be correct

In [70]:
df['default'].value_counts

<bound method IndexOpsMixin.value_counts of 0             no
1        unknown
2             no
3             no
4             no
          ...   
41183         no
41184         no
41185         no
41186         no
41187         no
Name: default, Length: 41188, dtype: object>

In [71]:
df['education'].value_counts()

university.degree      12168
high.school             9515
basic.9y                6045
professional.course     5243
basic.4y                4176
basic.6y                2292
unknown                 1731
illiterate                18
Name: education, dtype: int64

There are some columns with the value == unknown

In [72]:
df['month'].value_counts()

may    13769
jul     7174
aug     6178
jun     5318
nov     4101
apr     2632
oct      718
sep      570
mar      546
dec      182
Name: month, dtype: int64

There are no "January" or "February" in the "month" column of the data set